### PR TITLE
Adding DNF sort

### PR DESCRIPTION
**Pull request by kowshikaRani**
# Welcome to Dart community

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Dart files are placed inside an existing directory.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: dnfsort.dart [hunk 1]
```diff
@@ -0,0 +1,57 @@
+
+// Dutch National Flag (DNF) - It is a programming problem proposed by Edsger Dijkstra. The flag of the Netherlands consists of three colors: white, red, and blue. ... For DNF (Dutch National Flag), we sort an array of 0, 1, and 2's in linear time that does not consume any extra space.
+
+
+import 'package:test/test.dart';
+import 'dart:core';
+List sort012(List<int> a, int arrsize) {
```
Comment (akashgk) [inline dnfsort.dart:7]: Can you you use a better function name

File: dnfsort.dart [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  void main(List<String> args)
+  {
+    List<int> arr = [2,2,1,0,0,1];
+    int arrSize = arr.length;
+    sort012(arr, arrSize);
+    print("Array after seggregation ");
+    print("$arr");
+    test('test 1', () {
```
Comment (akashgk) [inline dnfsort.dart:42]: Add another test case for empty array and a test case handling the error

**Review by akashgk — CHANGES_REQUESTED** [review]

**Review by akashgkrishnan — APPROVED** [review]
